app/controllers/users_controller.rb

```diff
@@ -1,7 +1,11 @@
 class UsersController < ApplicationController
 
   def show
-    @user = current_user
+    if User.exists?(params[:id])
```

Comment:
Flip this around. Grab user and then check they are not `nil`. Otherwise we're doing 2 consecutive database queries.